pyiceberg/table/__init__.py

```diff
@@ -340,6 +341,86 @@ def set_properties(self, properties: Properties = EMPTY_DICT, **kwargs: Any) ->
         updates = properties or kwargs
         return self._apply((SetPropertiesUpdate(updates=updates),))
 
+    @deprecated(
+        deprecated_in="0.7.0",
+        removed_in="0.8.0",
+        help_message="Please use one of the functions in ManageSnapshots instead",
+    )
+    def add_snapshot(self, snapshot: Snapshot) -> Transaction:
+        """Add a new snapshot to the table.
+
+        Returns:
+            The transaction with the add-snapshot staged.
+        """
+        updates = (AddSnapshotUpdate(snapshot=snapshot),)
+        requirements = (AssertTableUUID(uuid=self._table.metadata.table_uuid),)
+
+        return self._apply(updates, requirements)
+
+    @deprecated(
+        deprecated_in="0.7.0",
+        removed_in="0.8.0",
+        help_message="Please use one of the functions in ManageSnapshots instead",
+    )
+    def set_ref_snapshot(
+        self,
+        snapshot_id: int,
+        parent_snapshot_id: Optional[int],
+        ref_name: str,
+        type: str,
+        max_ref_age_ms: Optional[int] = None,
+        max_snapshot_age_ms: Optional[int] = None,
+        min_snapshots_to_keep: Optional[int] = None,
+    ) -> Transaction:
+        """Update a ref to a snapshot.
+
+        Returns:
+            The transaction with the set-snapshot-ref staged
+        """
+        updates = (
+            SetSnapshotRefUpdate(
+                snapshot_id=snapshot_id,
+                ref_name=ref_name,
+                type=type,
+                max_ref_age_ms=max_ref_age_ms,
+                max_snapshot_age_ms=max_snapshot_age_ms,
+                min_snapshots_to_keep=min_snapshots_to_keep,
+            ),
+        )
+
+        requirements = (
+            AssertRefSnapshotId(snapshot_id=parent_snapshot_id, ref="main"),
+            AssertTableUUID(uuid=self.table_metadata.table_uuid),
```

Comment:
I think the `AssertTableUUID` is a very coarse way of checking if there is a conflict. The UUID is unique per commit, but there are many situations where the commits can be de-conflicted. For example, if you set table properties and you update a branch.

I noticed that in the PyIceberg code, and also in the open-api spec, the `type` is not properly defined. I took the liberty of creating a PR to fix that on the Iceberg side: https://github.com/apache/iceberg/pull/10423/